internal/log/slogger.go

```diff
@@ -0,0 +1,82 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"log/slog"
+)
+
+// custom slog.Level values.
+const (
+	LevelTrace = slog.Level(-8)
+	LevelDebug = slog.LevelDebug
+	LevelInfo  = slog.LevelInfo
+	LevelWarn  = slog.LevelWarn
+	LevelError = slog.LevelError
+	LevelFatal = slog.Level(9)
+)
+
+// string representation of custom slog.Level levels; defining them as constants is
+// recommended at: https://pkg.go.dev/log/slog#example-HandlerOptions-CustomLevels.
+const (
+	LevelTraceString = "trace"
+	LevelDebugString = "debug"
+	LevelInfoString  = "info"
+	LevelWarnString  = "warning"
+	LevelErrorString = "error"
+	LevelFatalString = "fatal"
+)
+
+// NewSlogger takes an io.Writer and returns a new logger of type slog.Logger.
+func NewSlogger(out io.Writer, level string) *slog.JSONHandler {
+	var slevel slog.Level
+	switch {
+	case level == LevelTraceString:
```

Comment:
this can be rewritten with a tagged switch:

```go
switch level {
	case LevelTraceString:
		slevel = LevelTrace
...
```

Reply:
Fixed.